engines/libpmem.c

```diff
@@ -212,12 +212,6 @@ static int fio_libpmem_close_file(struct thread_data *td, struct fio_file *f)
 	int ret = 0;
 
 	dprint(FD_IO, "DEBUG fio_libpmem_close_file\n");
-	dprint(FD_IO, "td->o.odirect %d\n", td->o.odirect);
-
-	if (!td->o.odirect) {
-		dprint(FD_IO,"pmem_drain\n");
-		pmem_drain();
-	}
```

Comment:
Out of curiosity do know why the original author might have believed it was?

Reply:
I'm not sure. In general you want your data to be persistent on the device (on NVDIMM), so maybe it was added just like that. It's not a mistake it is here, it's just seems redundant.
Anyway, I extended the commit msg a little bit, just FYI.

Reply:
Shouldn't we have pmem_drain() before the unmap() to ensure everything is persistent ?

Reply:
As I stated in the commit message:
- if we set the direct flag to 1, we use "PMEM_F_MEM_NONTEMPORAL", so non-temporal stores, so we don't need to worry about it (we avoid the need to flush processor caches)
- if we set the direct flag to 0, we simply don't care about this (at least in here, in the benchmarking. We rather don't want that in real-life scenarios)